ratpack-core/src/main/java/ratpack/health/HealthCheckHandler.java

```diff
@@ -0,0 +1,435 @@
+ * <ul>
+ *   <li>in parallel (if {@code concurrencyLevel=0})</li>
+ *   <li>in sequence (if {@code concurrencyLevel=1})</li>
+ *   <li>in groups (if {@code concurrencyLevel >= 2}). Groups are executed in sequence while promises inside group in parallel</li>
+ * </ul>
+ * <p>
+ * This handler should be bound to an application path, and most likely only for the GET method.
+ * <pre class="java-chain-dsl">
+ * import ratpack.health.HealthCheckHandler;
+ *
+ * chain.get("health-checks", new HealthCheckHandler());
+ * </pre>
+ * <p>
+ * The handler can render the result of all of the health checks or an individual health check, depending on the presence of a path token.
+ * The path token provides the name of the health check to render.
+ * If the path token is not present, all health checks will be rendered.
+ * The token name to use can be provided as the construction argument to this handler.
+ * The default token name is {@value #DEFAULT_NAME_TOKEN} and is used if the no-arg constructor is used.
+ * <p>
+ * If the token is present, the health check whose name is the value of the token will be rendered.
+ * If no health check exists by that name, the client will receive a {@code HTTP 404}.
+ * <p>
+ * When a single health check is selected (by presence of the path token)
+ * the {@link ratpack.health.HealthCheckResults} with one {@link ratpack.health.HealthCheck.Result}
+ * is {@link Context#render(Object) rendered}.
+ * When rendering all health checks a {@link ratpack.health.HealthCheckResults} is {@link Context#render(Object) rendered}.
+ * <p>
+ * The default {@link ratpack.health.HealthCheckResultsRenderer} is added to base registry. It renders in plain text.
+ * If you wish to change the output, to JSON for example, you can register your own renderer for {@link ratpack.health.HealthCheckResults}.
+ * See example in test cases {@code HealthCheckHandlerSpec}.
+ * <pre class="java">{@code
+ * import ratpack.exec.ExecControl;
+ * import ratpack.exec.Promise;
+ * import ratpack.guice.Guice;
+ * import ratpack.health.HealthCheck;
+ * import ratpack.health.HealthCheckHandler;
+ * import ratpack.health.HealthCheckResultsRenderer;
+ * import ratpack.test.embed.EmbeddedApp;
+ *
+ * import static org.junit.Assert.*;
+ *
+ * public class Example {
+ *   public static class FooHealthCheck implements HealthCheck {
+ *     public String getName() { return "foo"; }
+ *     public Promise<HealthCheck.Result> check(ExecControl execControl) throws Exception {
+ *       return execControl.promise(f -> {
+ *         f.success(HealthCheck.Result.healthy());
+ *       });
+ *     }
+ *   }
+ *
+ *   public static void main(String... args) throws Exception {
+ *     EmbeddedApp.of(s -> s
+ *       .registry(Guice.registry(b -> b
+ *         .bind(FooHealthCheck.class)
+ *         .bindInstance(HealthCheckResultsRenderer.class, new HealthCheckResultsRenderer())
+ *         .bindInstance(HealthCheckHandler.class, new HealthCheckHandler())
```

Comment:
I'd remove this from the example and just add a directly created instance below.